collections/_guides/embedding-a-smart-on-fhir-application.md

```diff
@@ -0,0 +1,182 @@
+---
+title: "Embedding a SMART app into Canvas"
+guide_for:
+- /sdk/handlers-applications/
+---
+
+Canvas' FHIR API supports the SMART launch sequence, and the Canvas SDK allows
+you to embed html content within the Canvas EMR. The combination of these
+capabilities is a frictionless experience when embedding a SMART app into
+Canvas. This example will show you how to register a SMART app's credentials
+and create an application to launch it from within a patient's chart.
+
+## In this guide you will learn how to:
+
+  - Configure credentials for a public OAuth application.
+  - Create an application to initiate the SMART launch sequence.
+  - Use application launch context to help form the SMART launch URL parameters.
+
+## What is a SMART app?
+
+SMART on FHIR applications adhere to a standards-based approach to
+authentication, authorization, and data access with the goal of being portable
+across EMR installations and vendors. Because SMART apps limit themselves to
+standards based data, the logic can be agnostic to the EMR it is running
+against. Vendor to vendor, site to site, FHIR is FHIR, that's how it works,
+right?
+
+## Where can I get a SMART app?
+
+We've provided an example SMART app
+[here](https://github.com/canvas-medical/example-smart-on-fhir-app). You can
+fork it and customize, or just use ours as-is. We use GitHub pages to host it
+at `https://canvas-medical.github.io/example-smart-on-fhir-app/`. Please
+ensure the client id [listed in
+`launch.html`](https://github.com/canvas-medical/example-smart-on-fhir-app/blob/205b14ed8d4f4d9c57fd11a26f9a22c800f38c29/launch.html#L15) matches the client id for the
+credentials you configure in the next step.
+
+## Configuring credentials
+
+The first step to embedding a SMART application is to configure OAuth
+credentials for it to use. Navigate to `<YOUR_CANVAS_URL>/auth/applications/register/`
+and choose `Public` client type, `Authorization code` grant type, the `RSA`
+algorithm, and set the redirect uri for your SMART application.
```

Comment:
the redirect uri in the example is a github.io page -- what is a more realistic example of where these applications would be hosted? would it be something like
https://<instance-name>.canvasmedical.com/plugin-io/api/PLUG-IN-NAME/my-app-route ? (This is for my own knowledge unless you think it makes sense to include as an example here)

Reply:
This guide is all about showing how an externally hosted application can be embedded into Canvas, so this is actually a realistic example. It being hosted by GitHub (more specifically just "not hosted by Canvas") is part of what we're showcasing.

Reply:
Got it, my brain is on Canvas hosted apps which is probably a different guide!

Reply:
I make my pitch for that alternative in the last section